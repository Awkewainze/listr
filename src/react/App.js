import React from 'react';
import './App.scss';
import Header from './common/Header';
import Footer from './common/Footer';
import { Switch, Route } from 'react-router-dom';
import ActiveListContainer from './list/ActiveListContainer';

function App() {
  return (
    <div className="App">
      <Header/>
      <Switch>
        <Route path="/" component={ActiveListContainer} />
      </Switch>
      <Footer/>
    </div>
  );
}

export default App;
