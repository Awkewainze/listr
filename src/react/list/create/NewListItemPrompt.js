import React from 'react';
import './NewListItemPrompt.scss';

//TODO add editable state here

const NewListItemPrompt = () => {
  const onSubmit = event => {
    //TODO only submit if clicked?
    //TODO validate input, add to list
    event.preventDefault();
  };

  return (
    <form className="NewListItemPrompt" onSubmit={onSubmit}>
      <input className="NewListItemPrompt__input" type="text" placeholder="...New item"/>
    </form>
  );
};

export default NewListItemPrompt;