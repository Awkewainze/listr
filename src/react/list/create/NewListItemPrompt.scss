.NewListItemPrompt {
  margin: 6px 0;
  border-bottom: 1px solid #afafaf;
  padding: 32px 20px 32px 92px;
  display: flex;

  &:hover {
    border-color: #7f7f7f;
  }

  .NewListItemPrompt__input, .NewlistItemPrompt__submit {
    font-size: 16px;
    outline: none;
    border: none;
  }

  .NewListItemPrompt__input {
    background-color: transparent;
    flex: 1;
  }

  .NewlistItemPrompt__submit {
    font-size: 16px;
    margin: 0 12px 0 24px;
    background-color: #09cb9f;
    color: #fff;
    padding: 12px;
    border-radius: 6px;
    cursor: pointer;

    &:hover {
      opacity: 0.9;
    }
  }
}