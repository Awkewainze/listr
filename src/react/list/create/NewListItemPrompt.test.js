import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewListItemPrompt from './NewListItemPrompt';

const onNewItem = jest.fn();

beforeEach(() => {
  onNewItem.mockReset();
});

it('matches snapshot', () => {
  const { container } = render(<NewListItemPrompt onNewItem={onNewItem} />);
  expect(container.firstChild).toMatchSnapshot();
});

it('does not trigger callback if value is empty', () => {
  const { container } = render(<NewListItemPrompt onNewItem={onNewItem}/>);
  fireEvent.submit(container.querySelector('form'));
  expect(onNewItem).toHaveBeenCalledTimes(0);
});

it('sends value to callback on submit', () => {
  const { container } = render(<NewListItemPrompt onNewItem={onNewItem}/>);
  const expectedText = 'oranges';
  fireEvent.change(container.querySelector('input'), { target: { value: expectedText } });
  fireEvent.submit(container.querySelector('form'));

  expect(onNewItem).toHaveBeenCalledTimes(1);
  expect(onNewItem).toHaveBeenCalledWith(expectedText);
});