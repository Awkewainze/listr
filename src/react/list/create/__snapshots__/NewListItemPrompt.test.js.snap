// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  class="NewListItemPrompt"
>
  <input
    class="NewListItemPrompt__input"
    placeholder="...New item"
    type="text"
    value=""
  />
  <button
    class="NewlistItemPrompt__submit"
    type="submit"
  >
    Submit
  </button>
</form>
`;
